astropy/coordinates/representation.py

```diff
@@ -422,19 +438,77 @@ class BaseRepresentation(BaseRepresentationOrDifferential):
     All representation classes should subclass this base representation class,
     and define an ``attr_classes`` attribute, an `~collections.OrderedDict`
     which maps component names to the class that creates them. They must also
-    define a ``to_cartesian`` method and a ``from_cartesian`` class method. By
-    default, transformations are done via the cartesian system, but classes
-    that want to define a smarter transformation path can overload the
-    ``represent_as`` method. If one wants to use an associated differential
-    class, one should also define ``unit_vectors`` and ``scale_factors``
-    methods (see those methods for details). Finally, classes can also define a
-    ``recommended_units`` dictionary, which maps component names to the units
-    they are best presented to users in (this is used only in representations
-    of coordinates, and may be overridden by frame classes).
+    define a ``_to_cartesian_helper`` method and a ``_from_cartesian_helper``
+    class method. By default, transformations are done via the cartesian
+    system, but classes that want to define a smarter transformation path can
+    overload the ``represent_as`` method. If one wants to use an associated
+    differential class, one should also define ``unit_vectors`` and
+    ``scale_factors`` methods (see those methods for details). Finally, classes
+    can also define a ``recommended_units`` dictionary, which maps component
+    names to the units they are best presented to users in (this is used only in
+    representations of coordinates, and may be overridden by frame classes).
     """
 
     recommended_units = {}  # subclasses can override
 
+    def __init__(self, *args, **kwargs):
+        # now handle any differentials passed in
+        differentials = kwargs.pop('differentials', None)
+
+        super(BaseRepresentation, self).__init__(*args, **kwargs)
+
+        # store as a private name so users know this is not settable
```

Comment:
If we really want to enforce non-setting of differentials, it may be best to put the validation logic in the property setter, and raise an exception if ```self._differentials``` exists.